datadog_checks_downloader/tests/local_http.py

```diff
@@ -0,0 +1,81 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import contextlib
+import logging
+import os
+import time
+from functools import partial
+from threading import Thread
+
+import requests
+import six
+
+if six.PY3:
+    from http.server import SimpleHTTPRequestHandler
+    from queue import Queue
+    from socketserver import TCPServer
+else:
+    from Queue import Queue
+    from SimpleHTTPServer import SimpleHTTPRequestHandler
+    from SocketServer import TCPServer
+
+_LOGGER = logging.getLogger(__name__)
+E2E_TESTS_DATA_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), "data")
+_DEFAULT_PORT = int(os.getenv("DATADOG_CHECKS_DOWNLOADER_E2E_TEST_HTTP_PORT", 8080))
+
+
+class _CustomTCPServer(TCPServer):
+    """A custom TCP server that reuses address."""
+
+    allow_reuse_address = True
+
+
+def _do_local_http_server(q, directory, port):
```

Comment:
```suggestion
def _do_local_http_server(queue, directory, port):
```
nit can we make this more descriptive

Reply:
Adjusted in 1f3ef7a9d8